compositor_render/src/scene/transition.rs

```diff
@@ -0,0 +1,96 @@
+use std::time::Duration;
+
+use super::{types::interpolation::InterpolationState, InterpolationKind};
+
+/// Similar concept to InterpolationState, but it represents a time instead.
+/// Values between 0 and 1 represent transition and larger than 1 post transition.
+///
+/// If interpolation_kind is linear then InterpolationState and TransitionProgress
+/// have the same numerical value.
+#[derive(Debug, Clone, Copy)]
+struct TransitionProgress(f64);
+
+#[derive(Debug, Clone)]
+pub(super) struct TransitionState {
+    /// Additional offset for transition. It is non zero if you want to start
+    /// a transition in the middle of the interpolation curve.
+    initial_offset: (TransitionProgress, InterpolationState),
+
+    // PTS of a first frame of transition.
+    start_pts: Duration,
+
+    /// Duration of the transition.
+    duration: Duration,
+
+    interpolation_kind: InterpolationKind,
+}
+
+pub(super) struct TransitionOptions {
+    pub duration: Duration,
+    pub interpolation_kind: InterpolationKind,
+}
+
+impl TransitionState {
+    pub fn from_previous_transition(
+        current_transition: Option<TransitionOptions>,
+        previous_transition: Option<TransitionState>,
```

Comment:
I don't have strong opinion on this but from the function's name alone I would expect that `previous_transition` is the first argument.

Reply:
I think arg order is fine here, if that `from` is misleading then we could change the name to just `new`.

Reply:
it seems a bit misleading to me as well